### ISSUE TITLE
Failed gRPC pipeline runs (when DagsterInstance can't be loaded) show wrong state in Dagit

### ISSUE THREAD

**Issue by kinghuang** [dagster-io/dagster#3016]
### Summary

I had several failed pipeline runs for repositories where Dagit is communicating with a remote gRPC server (they lead to #3013 and #3014). Dagit is showing these failed runs as **Starting…** under the `Timing` column in the `Runs` tabs.

### Steps to reproduce

1. Using Dagit, start a pipeline on a remote gRPC server that is known to fail. Use #3013 and #3014 for failure conditions.
2. Go to the `Runs` tab.

### What is the current *bug* behaviour?

Dagit shows the failed runs as **Starting…**.

### What is the expected *correct* behaviour?

Dagit should show the runs as failed.

### Relevant logs and/or screenshots

The lower four runs all failed. The top one completed successfully.

<img width="1252" alt="Screen Shot 2020-10-05 at 6 54 34 PM" src="https://user-images.githubusercontent.com/27340/95147155-2c25b400-073d-11eb-8098-8e68ac2208d6.png">

### Versions

```
root@994f48641753:/project# dagster --version
dagster, version 0.9.13
root@994f48641753:/project# dagit --version
dagit, version 0.9.13
```

**Comment by gibsondan** [comment]
What makes this tricky to solve in general is that both of the issues that caused this were problems loading the instance, which is also what we would use to mark the runs as failed (especially if the issue is specifically with loading the run storage as in #3013) . But we should still be able to figure out a solution here - the run launcher could mark it as failed if things fail before execution starts, for example.

**Comment by gibsondan** [comment]
This should be fixed in 0.10.0.
